core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -257,8 +257,8 @@ private[spark] class Executor(
           val serviceTime = System.currentTimeMillis() - taskStart
           val metrics = attemptedTask.flatMap(t => t.metrics)
           for (m <- metrics) {
-            m.executorRunTime = serviceTime
-            m.jvmGCTime = gcTime - startGCTime
+            m.incExecutorRunTime(serviceTime)
```

Comment:
Task GC time is measured by looking at ((total time spent in GC since process started) - (total time spent in GC since process started, snapshotted when task started)).  Incrementing every time we update the metric would end up double-counting certain time regions.  A few other metrics work this way as well.  If you do notice issues with the logic, it might be good to propose fixes in a separate JIRA, so that we can give them careful consideration on their own.